src/lib/treeviewNode.ts

```diff
@@ -0,0 +1,89 @@
+import { emptyBasicNode, BasicNode } from "./basicNode";
+import { Node } from "./node";
+
+type countAccessorFunction = (n: TreeviewNode) => number;
+type nodeData = {
```

Comment:
TreeView.data is overriding Node.data here. Perhaps it's better to specify a more general Node.data type? Also Node.Data for consistency.